assets/helpers/checkouts.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+// ----- Imports ----- //
+
+import { getQueryParameter } from 'helpers/url';
+import { getMinContribution, parseContribution, validateContribution } from 'helpers/contributions';
+import * as storage from 'helpers/storage';
+
+import type { Contrib } from 'helpers/contributions';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+
+
+// ----- Types ----- //
+
+export type PaymentMethod = 'DirectDebit' | 'PayPal' | 'Stripe';
+
+
+// ----- Functions ----- //
+
+function getAmount(contributionType: Contrib, countryGroup: CountryGroupId): number {
+
+  const contributionValue = getQueryParameter('contributionValue');
+
+  if (contributionValue !== null && contributionValue !== undefined) {
+    const parsed = parseContribution(contributionValue);
+
+    if (parsed.valid) {
+      const error = validateContribution(parsed.amount, contributionType, countryGroup);
+
+      if (!error) {
+        return parsed.amount;
+      }
+    }
+  }
+
+  return getMinContribution(contributionType, countryGroup);
```

Comment:
This is probably better from a user's perspective but I still think it'll be just as annoying as having any other value selected (if it's different from their own choice). 

I worry that masking the underlying problem means we won't be aware of existing or future browser-related issues. Can we add some logging somewhere to capture when this happens, along with whatever we can determine about the browser version?

Reply:
Yeah, I think it'll still be annoying if their value gets dropped, but as @johnduffell said, it's better to default to a lower amount than a higher one.

Yeah we could log this with sentry, good call!